tests/test_schunk.py

```diff
@@ -47,43 +47,74 @@ def test_schunk_numpy(contiguous, urlpath, mode, mmap_mode, cparams, dparams, nc
+            mmap_mode="w+" if mmap_mode is not None else None,
+            **storage,
+        )
+
+    assert schunk.urlpath == urlpath
+    assert schunk.contiguous == contiguous
+
+    for i in range(nchunks):
+        buffer = i * np.arange(chunk_len, dtype="int32")
+        nchunks_ = schunk.append_data(buffer)
+        assert nchunks_ == (i + 1)
+
+    if mode == "r":
+        if urlpath is not None:
+            schunk = blosc2.SChunk(chunksize=chunk_len * 4, mode=mode, mmap_mode=mmap_mode, **storage)
+        else:
+            return
+
+    for i in range(nchunks):
+        buffer = i * np.arange(chunk_len, dtype="int32")
+        bytes_obj = buffer.tobytes()
+        res = schunk.decompress_chunk(i)
+        assert res == bytes_obj
 
-        for i in range(nchunks):
-            buffer = i * np.arange(chunk_len, dtype="int32")
-            nchunks_ = schunk.append_data(buffer)
-            assert nchunks_ == (i + 1)
+        dest = np.empty(buffer.shape, buffer.dtype)
+        schunk.decompress_chunk(i, dest)
+        assert np.array_equal(buffer, dest)
 
-        for i in range(nchunks):
-            buffer = i * np.arange(chunk_len, dtype="int32")
-            bytes_obj = buffer.tobytes()
-            res = schunk.decompress_chunk(i)
-            assert res == bytes_obj
+        schunk.decompress_chunk(i, memoryview(dest))
+        assert np.array_equal(buffer, dest)
+
+        dest = bytearray(buffer)
+        schunk.decompress_chunk(i, dest)
+        assert dest == bytes_obj
 
-            dest = np.empty(buffer.shape, buffer.dtype)
-            schunk.decompress_chunk(i, dest)
-            assert np.array_equal(buffer, dest)
+    for i in range(nchunks):
+        schunk.get_chunk(i)
 
-            schunk.decompress_chunk(i, memoryview(dest))
-            assert np.array_equal(buffer, dest)
+    if nchunks >= 2:
+        assert schunk.cratio > 1
+        assert schunk.cratio == schunk.nbytes / schunk.cbytes
+    assert schunk.nbytes >= nchunks * chunk_len * 4
 
-            dest = bytearray(buffer)
-            schunk.decompress_chunk(i, dest)
-            assert dest == bytes_obj
+    blosc2.remove_urlpath(urlpath)
 
-        for i in range(nchunks):
-            schunk.get_chunk(i)
 
-        if nchunks >= 2:
-            assert schunk.cratio > 1
-            assert schunk.cratio == schunk.nbytes / schunk.cbytes
-        assert schunk.nbytes >= nchunks * chunk_len * 4
+@pytest.mark.parametrize(
+    "mode_write, mode_read, mmap_mode_write, mmap_mode_read",
+    [("w", "r", None, None), (None, None, "w+", "r")],
+)
+def test_schunk_ndarray(tmp_path, mode_write, mode_read, mmap_mode_write, mmap_mode_read):
+    urlpath = tmp_path / "test.b2nd"
 
-        blosc2.remove_urlpath(urlpath)
+    data = np.arange(2 * 10, dtype="int32")
+    blosc2.asarray(data, urlpath=urlpath, mode=mode_write, mmap_mode=mmap_mode_write)
+    with pytest.raises(ValueError, match="Cannot open an ndarray as a SChunk"):
```

Comment:
`ndarray` -> `NDArray` . The case is important here, most specially to distinguish from a NumPy ndarray.